dGame/dUtilities/SlashCommandHandler.cpp

```diff
@@ -75,32 +75,90 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& chat, Entity*
 
 void GMZeroCommands::Help(Entity* entity, const SystemAddress& sysAddr, const std::string args) {
 	std::ostringstream feedback;
-	if (args.empty()) {
-		feedback << "----- Commands -----";
-		for (const auto& [alias, command] : CommandInfos) {
-			// TODO: Limit displaying commands based on GM level they require
-			if (command.requiredLevel > entity->GetGMLevel()) continue;
-			LOG("Help command: %s", alias.c_str());
+	constexpr size_t pageSize = 10;
+
+	// Trim the args
+	std::string trimmedArgs = args;
+	trimmedArgs.erase(trimmedArgs.begin(), std::find_if_not(trimmedArgs.begin(), trimmedArgs.end(), [](unsigned char ch) {
+		return std::isspace(ch);
+		}));
+	trimmedArgs.erase(std::find_if_not(trimmedArgs.rbegin(), trimmedArgs.rend(), [](unsigned char ch) {
+		return std::isspace(ch);
+		}).base(), trimmedArgs.end());
+
+	// Check if the argument is a number (for pagination)
+	if (trimmedArgs.empty() || std::all_of(trimmedArgs.begin(), trimmedArgs.end(), ::isdigit)) {
```

Comment:
Try parse will fail if you try to parse a non number, the logic here can be simplified to that.  Stoi is not used anymore in dlu.